pkg/controller/vsphere/reconciler.go

```diff
@@ -633,6 +638,26 @@ func (vm *virtualMachine) getPowerState() (types.VirtualMachinePowerState, error
 	}
 }
 
+func (vm *virtualMachine) reconcileTags(ctx context.Context, session *session.Session, machine *machinev1.Machine) error {
+	if err := session.WithRestClient(vm.Context, func(c *rest.Client) error {
+		klog.Infof("%v: Reconciling attached tags", machine.GetName())
+
+		m := tags.NewManager(c)
+
+		clusterID := machine.Labels[machinev1.MachineClusterIDLabel]
+
+		if err := m.AttachTag(ctx, clusterID, vm.Ref); err != nil {
```

Comment:
Attach Tag requires the tag to already exist, is this something the installer is creating? Might be worth a comment if so?

Also, what happens if the Tag is already associated to the machine, will this succeed or error?

Reply:
Yes, it should be created by installer https://github.com/openshift/installer/blob/master/data/data/vsphere/main.tf#L49-L65
If the tag is already assigned it will succeed.